experimental/web_client/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#![feature(async_closure)]
+
+extern crate web_sys;
+
+use crate::proto::{UnaryRequest, UnaryResponse};
+use anyhow::Context;
+use async_trait::async_trait;
+use oak_functions_abi::proto::Response;
+use oak_remote_attestation_sessions::SessionId;
+use oak_remote_attestation_sessions_client::{GenericAttestationClient, UnaryClient};
+use prost::Message;
+use std::rc::Rc;
+use wasm_bindgen::prelude::*;
+
+mod grpc_web;
+
+#[cfg(test)]
+mod tests;
+
+mod proto {
+    #![allow(clippy::return_self_not_must_use)]
+    include!(concat!(env!("OUT_DIR"), "/oak.session.unary.v1.rs"));
+}
+
+// TODO(#1867): Add remote attestation support.
+const TEE_MEASUREMENT: &[u8] = br"Test TEE measurement";
+
+/// gRPC-web implementation of a [`UnaryClient`].
+struct GrpcWebClient {
+    uri: String,
+}
+
+impl GrpcWebClient {
+    pub fn create(uri: &str) -> Self {
+        Self {
+            uri: format!("{}/oak.session.unary.v1.UnarySession/Message", uri),
+        }
+    }
+}
+
+// Not marked as send, as the underlying client uses JavaScript APIs for
+// networking, which are not send.
+#[async_trait(? Send)]
```

Comment:
```suggestion
#[async_trait(?Send)]
```

`?Send` is usually kept together